src/cplscheme/CouplingData.cpp

```diff
@@ -129,26 +129,14 @@ std::vector<int> CouplingData::getVertexOffsets()
   return _mesh->getVertexOffsets();
 }
 
-void CouplingData::initializeExtrapolation()
-{
-  _extrapolation.initialize(getSize());
-  storeIteration();
-}
-
 void CouplingData::moveToNextWindow()
 {
-  _extrapolation.moveToNextWindow();
-  values() = _extrapolation.getInitialGuess();
-
-  this->setSampleAtTime(time::Storage::WINDOW_END, sample());
-}
-
-void CouplingData::storeExtrapolationData()
-{
-  const auto stamples = this->stamples();
-  PRECICE_ASSERT(stamples.size() > 0);
-  this->values() = stamples.back().sample.values;
-  _extrapolation.store(values());
+  if (this->timeStepsStorage().stamples().size() > 0) {
+    this->timeStepsStorage().move();
+    auto atEnd = this->timeStepsStorage().stamples().back();
+    PRECICE_ASSERT(math::equals(atEnd.timestamp, time::Storage::WINDOW_END));
+    _data->sample() = atEnd.sample;
+  }
 }
```

Comment:
Previously, this did not move the Storage of the coupling data, but computed the extrapolation (remember: for acceleration data) and stored it at the `WINDOW_END` of the Storage. Technically this is the same like moving, but the implementation is different.